src/service/timer-impl.js

```diff
@@ -132,14 +132,19 @@ export class Timer {
    * @param {number} delay
    * @param {?Promise<RESULT>|undefined} opt_racePromise
    * @param {string=} opt_message
+   * @param {boolean=} opt_resolveOnTimeout
    * @return {!Promise<RESULT>}
    * @template RESULT
    */
-  timeoutPromise(delay, opt_racePromise, opt_message) {
+  timeoutPromise(delay, opt_racePromise, opt_message, opt_resolveOnTimeout) {
```

Comment:
I'd rather we simply catch the `catch()` part at the call site.

Reply:
Thanks for the review. I'm happy to make this change. Just to be sure, `user().createError(opt_message || 'timeout')` doesn't report an error to ampproject.org, does it?

Reply:
Not if it's handled by the `catch()`.

Reply:
Changes to `timer-impl.js` removed as of e84ad45d321bc00ec7f9493ad4adb7d761f9e88a and `reject` is caught instead.